rfcs/0180-broken-package-removal.md

```diff
@@ -0,0 +1,91 @@
+---
+feature: broken-package-removal
+start-date: 2024-07-14
+author: Jörg "Mic92" Thalheim
+co-authors:
+shepherd-team:
+shepherd-leader:
+related-issues:
+---
+
+# Summary
+
+[summary]: #summary
+
+This RFCs defines under what conditions we remove broken and unmaintained leaf packages.
+The RFC does not have the ambition to define all reasons to remove a package but
+instead focuses on simple rules that can be automated.
+
+# Motivation
+
+[motivation]: #motivation
+
+Broken and unmaintained packages still cost us valuable time.
+Broken packages still have to be evaluated to some extent
+and unmaintained packages might still have to rebuild when one of their dependencies changes.
+Removing unmaintained/unused code hopefully increases the overall package quality and helps us
+to save resources.
+Because it is easy to add packages, it should be also easy to remove them, ideally automated.
+
+# Detailed design
+
+[design]: #detailed-design
+
+## Broken Packages
+
+Packages that are marked as broken unconditionally platforms should be removed after one full NixOS release cycle.
+If the package had dependencies, those dependencies are also marked as indirectly broken.
+If they are not functional without the broken package, they should be removed as well.
+
+## Unmaintained packages
+
+Packages with an empty maintainer field that does not have packages depending on it, should
+be removed after 2 months. Ideally we have an automation or semi-automation that creates pull requests for that (see future work).
```

Comment:
2 months is a bit too fast, I hope around 6 months

Reply:
It can be made in parallel with ZHF.

Reply:
To make it simpler, I unified the times for both cases.